woocommerce/woocommerce-gateway-stripe
# Add UPE React Settings wrapper

# Changes proposed in this Pull Request:

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #1680 

Description: This PR adds a React wrapper app to the Stripe checkout settings page. This serves as a starting point for adding React components to the settings page.

# Testing instructions

- Make sure to enable UPE preview with the following: 
```INSERT INTO `wp_options` (`option_id`, `option_name`, `option_value`, `autoload`) VALUES (NULL, '_wcstripe_feature_upe', '1', 'no') ON DUPLICATE KEY UPDATE `option_value`=VALUES(`option_value`);```
- With PR checked out, make sure to run webpack with `npm run start:webpack`
- Verify that you have three new files in the `build/` folder: `upe_settings.js`, `upe_settings.js.map`, `upe_settings.asset.php`.
- Go to the Checkout page: `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe`
- Verify that the `upe_settings.js` file is loaded on that page and that you have a child `<div>` element of the `<div id="wc-stripe-account-settings-container</div>` element.
- Check that no JS errors exist.

![Screen Shot 2021-08-19 at 10 40 09 AM](https://user-images.githubusercontent.com/6723003/130039739-d8bc93f3-1bd4-445b-b537-249c7e5150de.png)

-------------------

A basic smoketest will be performed by #1694 since there are no testing capabilities for React components at the moment.
